osu.Game.Rulesets.Osu/Skinning/Default/SpinnerRotationTracker.cs

```diff
@@ -107,9 +112,74 @@ public void AddRotation(float angle)
             }
 
             currentRotation += angle;
+
             // rate has to be applied each frame, because it's not guaranteed to be constant throughout playback
             // (see: ModTimeRamp)
-            drawableSpinner.Result.RateAdjustedRotation += (float)(Math.Abs(angle) * (gameplayClock?.GetTrueGameplayRate() ?? Clock.Rate));
+            double time = gameplayClock?.CurrentTime ?? Clock.CurrentTime;
+            double rate = gameplayClock?.GetTrueGameplayRate() ?? Clock.Rate;
+            float rateAdjustedRotation = (float)(angle * rate);
+
+            if (rate > 0)
+            {
+                int direction = Math.Sign(rateAdjustedRotation);
+
+                if (direction != 0 && direction != currentDirection)
+                {
+                    turningPoints.Add((time, currentDelta));
+                    currentDelta = 0;
```

Comment:
Is there a reason we can't just store the cumulative delta to `turningPoints`? If there is, it's not immediately obvious (so some kind of commenting explaining why on the relevant code would be appreciated).